crates/runtime/src/traphandlers.rs

```diff
@@ -167,6 +167,13 @@ cfg_if::cfg_if! {
                 } else if #[cfg(target_os = "macos")] {
                     let cx = &*(cx as *const libc::ucontext_t);
                     (*cx.uc_mcontext).__ss.__rip as *const u8
+                } else if #[cfg(target_os = "freebsd")] {
+                    let cx = &*(cx as *const libc::ucontext_t);
+                    if #[cfg(target_arch = "i386")] {
```

Comment:
I think this may not be valid syntax, you may want to double-check that this compiles perhaps?